x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTestHelper.java

```diff
@@ -242,41 +245,58 @@ public static CrossClusterAccessSubjectInfo randomCrossClusterAccessSubjectInfo(
     ) {
         try {
             // TODO add apikey() once we have querying-cluster-side API key support
-            final Authentication authentication = ESTestCase.randomFrom(
-                AuthenticationTestHelper.builder().realm(),
-                AuthenticationTestHelper.builder().internal(SystemUser.INSTANCE)
-            ).build();
+            final Authentication authentication = AuthenticationTestHelper.builder().realm().build();
             return new CrossClusterAccessSubjectInfo(authentication, roleDescriptorsIntersection);
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
+    public static CrossClusterAccessSubjectInfo randomCrossClusterAccessSubjectInfoForInternalUser(boolean emptyRoleDescriptors) {
```

Comment:
Nit: I think we can drop `random` from the method name since there isn't anything random in it.

Reply:
https://github.com/elastic/elasticsearch/pull/93808 will add apikeys here, so the method shall be random again. I'll coordinate the merge with @slobodanadamovic but I think it makes sense not to rename temporarily.

Reply:
Oh wait no my bad. I did not look closely enough at the diff -- this on is for the internal user so you're right. Will rename.